clash-lib/src/Clash/Netlist.hs

```diff
@@ -673,16 +667,54 @@ mkFunApp dstId fun args tickDecls = do
           (hWTysFiltered, argExprsFiltered) = unzip filteredTypeExprs
 
         dstHWty  <- unsafeCoreTypeToHWTypeM' $(curLoc) fResTy
-        env  <- Lens.use hdlDir
-        let modName = takeWhile (/= '.') (StrictText.unpack (nameOcc (varName fun)))
-
-        manFile <- case annM of
-          Just _  -> return (env </> ".." </> modName </> topName </> topName <.> "manifest")
-          Nothing -> return (env </> topName <.> "manifest")
-        Just man <- readMaybe <$> liftIO (readFile manFile)
-        instDecls <- mkTopUnWrapper fun annM man (dstId,dstHWty)
-                       (zip argExprsFiltered hWTysFiltered)
-                       tickDecls
+
+        -- TODO: The commented code fetches the function definition from the
+        --       set of global bindings and uses it to replicate the port names
+        --       of it. However, this does rely on the binding actually being
+        --       present in the binding map. This isn't the case, as only
+        --       the current top entity (and its dependencies, stopping at other
+        --       top entities) are present. We can't add the non-normalized
+        --       version, as this logic relies on 'splitArguments' having
+        --       fired. Adding normalized versions would create a dependency
+        --       between two top entities, defeating the ability to compile in
+        --       parallel.
+        --
+        --       One option is to split the normalization process into two
+        --       chunks: preprocessing (e.g., 'splitArguments') and actually
+        --       normalizing. This would ensure only minimal work is being done
+        --       serially.
+        --
+        --       The current workaround is to not rely on named arguments, using
+        --       positional ones instead when instantiating a top entity.
```

Comment:
Is the current workaround preferable to this, or is it just that this option is worth a PR by itself?

Reply:
It would be nice to use named arguments again, but it's going to take quite an annoying restructure of how we normalize. In other words, other PR seems like a good idea to me!